cfgov/cfgov/settings/base.py

```diff
@@ -753,3 +721,84 @@
+# - OIDC_OP_TOKEN_ENDPOINT: token endpoint (/token)
+# - OIDC_OP_USER_ENDPOINT: userinfo endpoint (/userinfo)
+# - OIDC_OP_JWKS_ENDPOINT: JWKS endpoint (alternative to OIDC_RP_IDP_SIGN_KEY)
+#
+# Optional environment variables:
+# - OIDC_RP_IDP_SIGN_KEY: The key (PEM) to sign ID tokens when
+#                         OIDC_RP_SIGN_ALGO is RS256 (default: None)
+# - OIDC_ADMIN_GROUP: The group claim for admins
+# - OIDC_USER_GROUP: The group claim for regular users
+#
+# See the mozilla-django-oidc documentation for more details about the
+# settings below:
+# https://mozilla-django-oidc.readthedocs.io/en/stable/settings.html
+ENABLE_SSO = os.environ.get("ENABLE_SSO") == "True"
+if ENABLE_SSO:
+    INSTALLED_APPS += ("mozilla_django_oidc",)
+
+    # Add our OIDC authentication backend, a subclass of
+    # mozilla_django_oidc.auth.OIDCAuthenticationBackend
+    AUTHENTICATION_BACKENDS += ("login.auth.CFPBOIDCAuthenticationBackend",)
+
+    # Add OIDC middleware that refreshes sessions from the provider
+    MIDDLEWARE += ("mozilla_django_oidc.middleware.SessionRefresh",)
+
+    # Configure login/out URLs for OIDC
+    LOGIN_URL = "oidc_authentication_init"
+    LOGIN_REDIRECT_URL = reverse_lazy("wagtailadmin_home")
+    LOGOUT_REDIRECT_URL = "/"
+    ALLOW_LOGOUT_GET_METHOD = True
+
+    # This OIDC client's id and secret
+    OIDC_RP_CLIENT_ID = os.environ["OIDC_RP_CLIENT_ID"]
+    OIDC_RP_CLIENT_SECRET = os.environ["OIDC_RP_CLIENT_SECRET"]
+
+    # The OIDC provider's signing algorithms and key/key endpoint
+    OIDC_RP_SIGN_ALGO = os.environ["OIDC_RP_SIGN_ALGO"]
+    OIDC_RP_IDP_SIGN_KEY = os.environ.get("OIDC_RP_IDP_SIGN_KEY")
+    OIDC_OP_JWKS_ENDPOINT = os.environ.get("OIDC_OP_JWKS_ENDPOINT")
+
+    # OIDC provider endpoints
+    OIDC_OP_AUTHORIZATION_ENDPOINT = os.environ[
+        "OIDC_OP_AUTHORIZATION_ENDPOINT"
+    ]
+    OIDC_OP_TOKEN_ENDPOINT = os.environ["OIDC_OP_TOKEN_ENDPOINT"]
+    OIDC_OP_USER_ENDPOINT = os.environ["OIDC_OP_USER_ENDPOINT"]
+
+    # For users created just-in-time, assign a username based on the
+    # username portion of their email address.
+    OIDC_USERNAME_ALGO = "login.auth.username_from_email"
+
+    # For each OIDC claim we get we might want to perform some action to
+    # map the claim to something in Django or Wagtail.
+    OIDC_CLAIMS_PROCESSORS = {
+        "roles": "login.auth.process_roles_claim",
+        "given_name": "login.auth.process_given_name_claim",
+        "family_name": "login.auth.process_family_name_claim",
+    }
+
+    # Now we do some role/group-mapping for admins and regular users
+    # Upstream "role" for users who get is_superuser
+    OIDC_OP_ADMIN_ROLE = os.environ.get("OIDC_OP_ADMIN_ROLE")
+
+    # Upstream "role" assigned for regular users
+    OIDC_OP_USER_ROLE = os.environ.get("OIDC_OP_USER_ROLE")
+
+    # Default group that regular users get assigned to
+    SSO_DEFAULT_GROUP = os.environ.get("SSO_DEFAULT_GROUP", "All Users")
```

Comment:
That's actually intentional in `.env_SAMPLE` that it not fall-back, because for local dev the roles/groups stuff isn't testable against the Mozilla test provider.

However, I can work with the suggestion here.